### ISSUE TITLE
SCT gets stuck when trying to run a test with compile errors

### ISSUE THREAD

**Issue by HurryStarfish** [GWRon/SimpleCodeTester#8]
When SCT is running tests and encounters a test that causes compile errors, SCT gets stuck. The last thing printed on the colsole is the result of the previous test, after which the SCT process will fully take up one CPU core. Nothing then happen anymore, leaving no option but to kill the process.

**Comment by GWRon** [comment]
Hmm, do you have a faulty example?

SCT waits for a process as long as a process is "alive" - as returned by `fdProcessStatus(handle)` (from pub.mod/freeprocess.mod).
Aside of that it tries to read from the standardIO and errorIO pipe as long as there is something to read there (read: "has a newline at the end"). Once the process finished there might be something left in the readbuffer - which is read too. 
As long as either the pipes or the buffer have something to read, a "non alive"-process (aka "finshed" according to fdProcess) is kept being interpreted as "still alive".

**Comment by HurryStarfish** [comment]
Ohh sorry, I just realized the problem. I was thrown off by the fact that SCT is busy-waiting (I feel like it shouldn't be consuming a CPU core while just waiting for bcc/bmk to terminate), but it getting stuck was entirely my fault. Turns out I've been running the tests with a bcc version that was set up to go into a debug stop on error, so the process would never end, just sit there waiting for user input forever.
